svea_vision/scripts/static_image_publisher.py

```diff
@@ -0,0 +1,71 @@
+#! /usr/bin/env python3
+
+import os
+import cv2
+import random
+
+import rospy
+from cv_bridge import CvBridge
+from sensor_msgs.msg import Image
+
+def load_param(name, value=None):
+    if value is None:
+        assert rospy.has_param(name), f'Missing parameter "{name}"'
+    return rospy.get_param(name, value)
+
+class PublishImage:
+
+    def __init__(self):
+        try:
+            # Initialize node
+            rospy.init_node('static_image_publisher', anonymous=True)
+            
+            # Parameters
+            self.image_topic = load_param('~image_topic', 'static_image')
+            self.image_path = load_param('~image_path') # Path to a single image or a directory of images
+            self.rate = load_param('~rate', 30)
+
+            # CV Bridge
+            self.cv_bridge = CvBridge()
+
+            # Publisher
+            self.img_pub = rospy.Publisher(self.image_topic, Image, queue_size=1)
+
+        except Exception as e:
+            # Log error
+            rospy.logerr(e)
+
+        else:
+            # Log status
+            rospy.loginfo('{} node initialized.'.format(rospy.get_name()))
+
+    def run(self):
```

Comment:
In this case it might even be worse, or?

Reply:
> If exception then the node will spin (still alive) but certain functions/features might not work as expected.

At some point I realized this and hence started adding the extra shutdown signal to the `init` block, but I missed this one. I will update this if you're good with this change while catching the exception:

```
except Exception as e: 
     # Log error 
     rospy.logfatal("{}: {}".format(rospy.get_name(), e)) 
     rospy.signal_shutdown("Initialization failed: {}".format(e)) 
```

This also gives more control and a standard way to exit the node if the params are not as expected, for example:
https://github.com/KTH-SML/svea_vision/blob/093d29a347c1efdeecf4ab8742f484c8768f7d19/svea_vision/scripts/segment_anything.py#L109-L110

Reply:
> In general, this also gives more control and a standard way to exit the node if the params are not as expected

I do agree with this, that is nice. I guess this just tingles some dark, uncomfortable part of my brain when I see the try-except in an `__init__`. Not that I have anything against it in the PR, I only mean generally speaking.

Perhaps a good paradigm for ROS is to have the try-except at top-level?
```
class Node:
    def __init__(self): # initialization that can raise exceptions
    def run(self): # application code

if __name__ == '__main__':
    try:
        node = Node()
    except Exception as e:
        rospy.logfatal(...)
        rospy.signal_shutdown(...)
    else:
        node.run()
```